.github/workflows/release.yaml

```diff
@@ -241,3 +247,33 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           GPG_FINGERPRINT: ${{ steps.import_gpg.outputs.fingerprint }}
 
+  # Bump the provider version to the next dev
+  bump-to-dev-version:
+    needs: [ "goreleaser" ]
+    if: ${{ needs.goreleaser.outputs.status != 'failure' }}
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ssh-key: ${{secrets.DEPLOY_SSH_PRIVATE_KEY }}
+      - name: Bump project version in go code
+        id: bump-dev
+        run: |
+          version="${{ inputs.version }}"
+          
+          # Bump one patch up and add '-dev' suffix
+          dev="$(echo "${version}" | awk -F. '{$NF = $NF + 1;} 1' | sed 's/ /./g')-dev"
```

Comment:
Seems a bit hacky!
Maybe use ldflags for go build instead?

Reply:
example: https://github.com/ClickHouse/data-plane-application/blob/main/dataplanectl/Makefile#L51

Reply:
Also, worth adding git commit version IMO. With ldflags it is easier to pass multiple params than adding sed commands

Reply:
Man, I have been doing go for years and never knew this! Thanks!